lib/navigation/preprocess_keyence.cpp

```diff
@@ -2,24 +2,43 @@
 
 namespace hyped::navigation {
 
-KeyencePreprocessor::KeyencePreprocessor()
+KeyencePreprocessor::KeyencePreprocessor(hyped::core::ILogger &logger) : log_(logger)
 {
-  // TODOLater: implement
+  // Currently no extra parameters so constructor is empty
+  has_keyence_disagreed_ = false;
 }
 
-SensorChecks checkKeyenceAgrees(const core::KeyenceData &keyence_data)
+SensorChecks KeyencePreprocessor::checkKeyenceAgrees(const core::KeyenceData &keyence_data)
 {
   /*
    TODOLater: implement:
    roughly:
+    if
    - if keyence data disagrees and does_keyence_agree_ is false, fail state, return kUnacceptable
+
    - if keyence data disagrees and does_keyence_agree_ is true, set does_keyence_agree_ false return
    kAcceptable
    - if keyence data agrees and does_keyence_agree_ is false, et does_keyence_agree_ true return
    kAcceptable
    - if keyence data agrees and does_keyence_agree_ is true, all is good in the world, return
    kAcceptable.
   */
+  bool keyence_data_disagrees = false;
```

Comment:
comment what the purpose of this is probably (we both know because we fully understand the desired functionality but someone looking at this fresh might not know).